### PR TITLE
ci: integration tests with multiple lua versions on multiple platforms

### PR DESCRIPTION
**Pull request by mrcjkb**
Stacked on #228 (so we can run the busted integration test with multiple lua versions).

### CODE REVIEW

**Comment by mrcjkb** [comment]
This may be blocked by 
- #213

**Comment by mrcjkb** [comment]
Looks like we have a linker error on `aarch64-darwin` when installing `busted`:

```
called `Result::unwrap()` on an `Err` value: InstallError(BuildError(CommandFailure { status: ExitStatus(unix_wait_status(256)), stdout: "", stderr: "Undefined symbols for architecture arm64:\n  \"_luaL_argerror\"
# ...
```

**Comment by mrcjkb** [comment]
ah, what are these jj shenanigans?

File: rocks-lib/src/operations/install.rs [hunk 1]
```diff
@@ -26,11 +26,16 @@ use super::SearchAndDownloadError;
 #[derive(Error, Debug)]
 #[error(transparent)]
 pub enum InstallError {
+    #[error(transparent)]
     SearchAndDownloadError(#[from] SearchAndDownloadError),
+    #[error(transparent)]
     LuaVersionError(#[from] LuaVersionError),
+    #[error(transparent)]
     LuaVersionUnset(#[from] LuaVersionUnset),
+    #[error(transparent)]
```
Comment (vhyrro) [inline rocks-lib/src/operations/install.rs:35]: nitpick: because of the top-level `#[error(transparent)]` these extra statements here are redundant

**Review by vhyrro — APPROVED** [review]

File: rocks-lib/src/operations/install.rs (hunk 1 above)
Comment (mrcjkb) [reply rocks-lib/src/operations/install.rs:34]: Removed the top-level one, because it doesn't apply to all constructors anymore.

**Pull request merged**